crates/sui-framework/sources/display.move

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Defines a Display struct which defines the way an Object
+/// should be displayed. The intention is to keep data as independent
+/// from its display as possible, protecting the development process
+/// and keeping it separate from the ecosystem agreements.
+///
+/// Each of the fields of the Display object should allow for pattern
+/// substitution and filling-in the pieces using the data from the object T.
+///
+/// More entry functions might be added in the future depending on the use cases.
+module sui::display {
+    use sui::publisher::{is_package, Publisher};
+    use sui::tx_context::TxContext;
+    use std::string::{String, utf8};
+    use sui::vec_map::{Self, VecMap};
+    use sui::object::{Self, ID, UID};
+    use sui::transfer;
+    use sui::event;
+    use std::vector;
+
+    /// For when T does not belong to package in PublisherCap.
+    const ENotOwner: u64 = 0;
+
+    /// For when vectors passed into one of the multiple insert functions
+    /// don't match in their lengths.
+    const EVecLengthMismatch: u64 = 1;
+
+    /// The Display object. Defines the way an object should be
```

Comment:
```suggestion
    /// The Display object. Defines how *every* object of type `T` should be
```

, if this is accurate. I think we need to be very clear that there is one `Display`/type rather than one `Display`/object.